### PR TITLE
Refactor Azure response stripping to use configurable field paths

### PR DESCRIPTION
**Pull request by abdallahsamabd**
## Description

Azure OpenAI responses include provider-specific fields (`content_filter_results`, `prompt_filter_results`) that leak through to clients. The previous implementation (PR #57) hardcoded the stripping logic with explicit `if` checks and `delete` calls for each known field.

This PR introduces a **general-purpose `ResponseFieldStripper`** under `pkg/plugins/api-translation/translator/` that any translator can reuse. The stripping logic is data-driven and configurable via dot-separated field paths.

**Implementation:**

- `translator/response_field_stripper.go` — a shared `ResponseFieldStripper` struct with `NewResponseFieldStripper(fieldsToStrip []string)` and a `Strip(body)` method. Lives in the `translator` package so all providers can import it.
- `azure/azure_openai.go` — the Azure translator holds a `*translator.ResponseFieldStripper` and delegates response stripping to it via `t.stripper.Strip(body)`.
- `translator/response_field_stripper_test.go` — tests for the generic stripping mechanism (custom paths, nested maps, nil/empty paths, mixed array types, edge cases).
- `azure/azure_openai_test.go` — Azure-specific tests (default fields, streaming chunks, live mock integration).

**Path format conventions:**
- `"prompt_filter_results"` — deletes a top-level key
- `"choices[].content_filter_results"` — iterates an array and deletes from each element
- `"metadata.debug_info"` — navigates into a nested object and deletes the leaf key

**Default Azure configuration:**
var DefaultResponseFieldsToStrip = []string{
"prompt_filter_results",
"choices[].content_filter_results",
}

**Reusability:** Any future translator (e.g., Bedrock) can hold a `*translator.ResponseFieldStripper` and call `Strip()` with its own provider-specific field paths — no code duplication needed.

Addresses #67.

## Summary by CodeRabbit

* **Refactor**
  * Introduced a configurable response-field filtering system and integrated it into the Azure translation pathway to standardize returned responses.

* **Tests**
  * Added comprehensive tests covering nested paths, array element handling, mixed element types, no-op scenarios, and various edge cases.

### CODE REVIEW

File: pkg/plugins/api-translation/translator/azure/azure_openai.go [hunk 1]
```diff
@@ -74,29 +93,88 @@ func (t *AzureOpenAITranslator) TranslateRequest(body map[string]any) (map[strin
...
+type fieldSegment struct {
+	key     string
+	isArray bool // true when the segment had a "[]" suffix
+}
+
+// fieldPath is a parsed sequence of segments.
+type fieldPath []fieldSegment
+
+// parseFieldPaths converts raw dot-separated path strings into parsed fieldPaths.
+func parseFieldPaths(raw []string) []fieldPath {
```
Comment (nirrozenbaum) [inline pkg/plugins/api-translation/translator/azure/azure_openai.go]: @abdallahsamabd can we implement a general purpose `ResponseFieldStripper` struct under `/pkg/plugins/api-translation/translator/response_field_stripper.go`?

This seems like it should be reusable by other translators as well.
each translator may have it own fields to strip, but the mechanism should be the same.

Comment (abdallahsamabd) [reply pkg/plugins/api-translation/translator/azure/azure_openai.go]: done

File: pkg/plugins/api-translation/translator/response_field_stripper.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	}
+}
+
+// Strip walks the configured field paths and removes matching fields from the body.
+// Returns the mutated body and true if at least one field was removed.
+// If nothing was removed, returns nil and false.
+func (s *ResponseFieldStripper) Strip(body map[string]any) (map[string]any, bool) {
+	mutated := false
+	for _, fp := range s.fieldPaths {
+		if stripField(body, fp, 0) {
```
Comment (nirrozenbaum) [inline pkg/plugins/api-translation/translator/response_field_stripper.go:49]: where do you split the string based on `.`?

Comment (abdallahsamabd) [reply pkg/plugins/api-translation/translator/response_field_stripper.go:49]: The dot-splitting happens in parseFieldPaths(), which is called once at construction time — not inside Strip().

image

Comment (nirrozenbaum) [reply pkg/plugins/api-translation/translator/response_field_stripper.go:49]: thanks. somehow missed it.

File: pkg/plugins/api-translation/translator/azure/azure_openai.go [hunk 3]
```diff
@@ -28,18 +28,38 @@ const (
...
-func NewAzureOpenAITranslator() *AzureOpenAITranslator {
-	return &AzureOpenAITranslator{}
+// NewAzureOpenAITranslator initializes a new AzureOpenAITranslator.
+// responseFieldsToStrip defines which provider-specific fields to remove from responses,
+// using dot-separated paths with "[]" to denote array traversal.
+// Pass nil or an empty slice for no stripping (no-op response translation).
+func NewAzureOpenAITranslator(responseFieldsToStrip []string) *AzureOpenAITranslator {
+	return &AzureOpenAITranslator{
+		stripper: translator.NewResponseFieldStripper(responseFieldsToStrip),
+	}
```
Comment (nirrozenbaum) [inline pkg/plugins/api-translation/translator/azure/azure_openai.go]: can we remove the arg from the function and use the default value inside it?

```suggestion
func NewAzureOpenAITranslator() *AzureOpenAITranslator {
    responseFieldsToStrip := []string{"prompt_filter_results","choices[].content_filter_results"}
	return &AzureOpenAITranslator{
		stripper: translator.NewResponseFieldStripper(responseFieldsToStrip),
	}
```

File: pkg/plugins/api-translation/translator/azure/azure_openai.go [hunk 4]
```diff
@@ -28,18 +28,38 @@ const (
...
+//
+// Examples:
+//
+//	"prompt_filter_results"            → deletes the top-level key
+//	"choices[].content_filter_results" → iterates choices and deletes from each element
+var DefaultResponseFieldsToStrip = []string{
+	"prompt_filter_results",
+	"choices[].content_filter_results",
+}
+
```
Comment (nirrozenbaum) [inline pkg/plugins/api-translation/translator/azure/azure_openai.go]: and if we update the new func we can remove this

**Comment by nirrozenbaum** [comment]
@abdallahsamabd thanks for the PR. 
overall looks very good. 
I left minor comments, other than that it's ready for merge.

File: pkg/plugins/api-translation/translator/azure/azure_openai.go (hunk 3 above)
Comment (abdallahsamabd) [reply pkg/plugins/api-translation/translator/azure/azure_openai.go]: done

File: pkg/plugins/api-translation/translator/azure/azure_openai.go [hunk 5]
```diff
@@ -31,15 +31,23 @@ const (
 // compile-time interface check
 var _ translator.Translator = &AzureOpenAITranslator{}
 
-// NewAzureOpenAITranslator initializes a new AzureOpenAITranslator and returns its pointer.
+// NewAzureOpenAITranslator initializes a new AzureOpenAITranslator.
```
Comment (nirrozenbaum) [inline pkg/plugins/api-translation/translator/azure/azure_openai.go]:
```suggestion
// NewAzureOpenAITranslator initializes a new AzureOpenAITranslator and returns its pointer.
```

**Review by nirrozenbaum — APPROVED** [review]
/lgtm
/approve
